File: base/SavingsAccount.java
public class SavingsAccount {

	private double balance;
	private double interestRate;
	private int time;


	public SavingsAccount(double balance, double interestRate, int time) {

		super();
		this.balance = balance;
		this.interestRate = interestRate;
		this.time = time;

	}


	//getters

	public double getBalance() {

		return balance;

	}


	public double getInterestRate() {

		return interestRate;

	}


	//setters

	public void setInterestRate(double interestRate) {

		this.interestRate = interestRate;

	}


	public void setBalance(double balance) {

		this.balance = balance;

	}


	//functions

	public void deposit2(double balance) {

		this.balance = this.balance + balance;

	}


	public void withdraw2(double balance) {

		if(this.balance < balance) {

			System.out.println("Insuficient funds for this transaction.");

		} else {

			this.balance = this.balance - balance;

		}

	}


	public double calculateInterest(int time) {

		double total;

		total = this.balance * (1 + (interestRate*time));

		return total;

	}


	public void addInterest(int time) {

		this.balance = this.balance + calculateInterest(time);

	}





}

File: blackbelt/savingsAccount.cpp
#include "bankAccount.h"
#include "savingsAccount.h"
#include <iostream>
#include <string>

SavingsAccount::SavingsAccount() {

}

SavingsAccount::SavingsAccount(double balance, double interestRate) {
	SavingsAccount::balance = balance;
	SavingsAccount::interestRate = interestRate;
}

SavingsAccount::~SavingsAccount() {

}


//getters
double SavingsAccount::getBalance() {
	return SavingsAccount::balance;
}

double SavingsAccount::getInterestRate() {
	return SavingsAccount::interestRate;
}

int SavingsAccount::getTime() {
	return SavingsAccount::time1;
}



//setters
void SavingsAccount::setBalance(double balance) {
	SavingsAccount::balance = balance;
}

void SavingsAccount::setInterestRate(double interestRate) {
	SavingsAccount::interestRate = interestRate;
}

void SavingsAccount::setTime(int time1) {
	SavingsAccount::time1 = time1;
}




double SavingsAccount::calculateInterest(int time1) {

	double amount;

	amount = SavingsAccount::balance * (1 + (SavingsAccount::interestRate*time1));

	return amount;

}

void SavingsAccount::deposit(double balance) {

	SavingsAccount::balance = SavingsAccount::balance + balance;

}

void SavingsAccount::withdraw(double balance) {

	if(SavingsAccount::balance < balance) {

		std::cerr << "Insufficient funds!" << std::endl << std::endl;

	} else {

		SavingsAccount::balance = SavingsAccount::balance - balance;

	}

}

void SavingsAccount::addInterest(int time1) {

	SavingsAccount::balance = SavingsAccount::balance + calculateInterest(time1);

}

